src/default-directory-provider.coffee

```diff
@@ -42,3 +42,17 @@ class DefaultDirectoryProvider
   # * `null` if the given URI is not compatibile with this provider.
   directoryForURI: (uri) ->
     Promise.resolve(@directoryForURISync(uri))
+
+  # Public: Normalizes path.
+  #
+  # * `uri` {String} The path that should be normalized.
+  #
+  # Returns a {String} with normalized path.
+  normalizePath: (uri) ->
+    # Normalize disk drive letter on Windows to avoid opening two buffers for the same file
+    pathWithNormalizedDiskDriveLetter =
+      if process.platform is 'win32' and matchData = uri.match(/^([A-Za-z]):/)
```

Comment:
Seems redundant to include A-Z - if it's already upper-cased we don't need to process it.